### ISSUE TITLE
Build failed when I select opecv_world

### ISSUE THREAD

**Issue by Bahramudin** [opencv/opencv#8765]
##### System information (version)

- OpenCV = 3.2
- Operating System / Platform = Windows 64 Bit
- Compiler = Visual Studio 2017
##### Detailed description
When I try to build opencv from the source it works well, but when I select opencv_world, then it fails to build in Visual Studio 2017 (13 success, 62 failed), in CMake everything goes well. What can be the reason and how to build opencv with world. Thanks!

**Comment by alalek** [comment]
Compare your logs with "winpack" logs from [here](http://pullrequest.opencv.org/buildbot/waterfall?tag=pack-master) (MSVS2015).

**Comment by Bahramudin** [comment]
@alalek It is the screen shot when I build Install (not build all). I think there is something that needs to be configured in CMake?
![ss](https://cloud.githubusercontent.com/assets/13574895/26298319/67b621da-3f09-11e7-909f-a7ee230f415f.png)

**Comment by alalek** [comment]
Your log blames code from opencv_dnn module (from opencv_contrib). Try to disable it if you don't use them (via `cmake -DBUILD_opencv_dnn=OFF`).

**Comment by Bahramudin** [comment]
@alalek I have OFF the opencv_dd, but the problem still present, why opencv_world cannot be built, but when I download the prebuild package there is a .dll file which OK, I also want something like that, but it is not working.

**Comment by Bahramudin** [comment]
@alalek I found the problem, it was because of contrib_module (opencv_extra_modules). If I do not set the path to contrib_modules, then everything goes well. Now, what should we do with this problem, how we can build the OpenCV world with its extra modules?

**Comment by nishant07** [comment]
Hi @Bahramudin ,
Were you able to build successfully lately? If yes, discard rest of the comment.
If not,
Which version of opncv_contrib you are using? If it's 3.3.0 rc, check this [issue ](https://github.com/opencv/opencv_contrib/issues/1273) that I have submitted. It's highly likely that it's the same issue which I faced since I can see `M_PI undeclared identifier` error in your error log. Try the work around I suggested and check if it builds successfully.
